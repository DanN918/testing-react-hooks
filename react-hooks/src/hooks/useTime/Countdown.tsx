//
// Sample usage
//
import React from 'react';
import { DateTime } from 'luxon';
import { useTime } from '.';

interface ICountdownProps {
    end: DateTime,
}

export const Countdown: React.FC<ICountdownProps> = ({ end }: ICountdownProps) => {
    const now = useTime(200); // this countdown will queue a re-render every 200ms.
    // (it will try to update every 200ms)

    // Handling the out-of-time case (countdown finished!)
    if (now > end) {
        return (
            <h1>It is finished!</h1>
        );
    }

    // Luxon `DateTime`: https://moment.github.io/luxon/docs/class/src/datetime.js~DateTime.html#instance-method-diff
    const diff = end.diff(now);
    // Luxon `Duration`: https://moment.github.io/luxon/docs/class/src/duration.js~Duration.html#instance-method-toFormat
    const formattedDuration = diff.toFormat('hh:mm:ss');

    return (
        <h1>This app will self-destruct in {formattedDuration}</h1>
    );
}
