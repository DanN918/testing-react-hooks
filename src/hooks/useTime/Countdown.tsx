//
// Sample usage
//
import React from 'react';
import { DateTime } from 'luxon';
import { useTime, IUseTimeOptions } from '.';

interface ICountdownProps {
    end: DateTime,
    _useTimeOptions: IUseTimeOptions,
}

export const Countdown: React.FC<ICountdownProps> = ({ end, _useTimeOptions }: ICountdownProps) => {
    const now: DateTime = useTime(200, _useTimeOptions); // this countdown will queue a re-render every 200ms.
    // (it will try to update every 200ms)

    // Handling the out-of-time case (countdown finished!)
    if (now > end) {
        return (
            <h1>It is finished!</h1>
        );
    }

    // Luxon `DateTime`: https://moment.github.io/luxon/docs/class/src/datetime.js~DateTime.html#instance-method-diff
    const diff = end.diff(now);
    // Luxon `Duration`: https://moment.github.io/luxon/docs/class/src/duration.js~Duration.html#instance-method-toFormat
    const formattedDuration = diff.toFormat('hh:mm:ss');

    return (
        <h1>This app will self-destruct in {formattedDuration}</h1>
    );
}
