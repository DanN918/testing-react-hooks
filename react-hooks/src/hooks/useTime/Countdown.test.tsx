import React from 'react';
import renderer from 'react-test-renderer';

import { Countdown } from './Countdown';
import { DateTime } from 'luxon';

describe('Countdown', () => {
    it('should match snapshot', () => {
        const end = DateTime.local().plus({ seconds: 1.1 });
        const s = renderer.create(<Countdown end={end} />);

        expect(s.toJSON()).toMatchSnapshot();
    });

    // TODO(james.fulford): test component time functionality
});
