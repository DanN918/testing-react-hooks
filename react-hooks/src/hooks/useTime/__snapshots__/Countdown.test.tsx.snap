// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countdown should match snapshot 1`] = `
<h1>
  This app will self-destruct in 
  00:00:01
</h1>
`;
